sdk/test-utils/recorder/src/sanitizer.ts

```diff
@@ -1,230 +1,305 @@
-      await Promise.all(
-        options.connectionStringSanitizers.map((replacer) =>
-          this.addConnectionStringSanitizer(replacer.actualConnString, replacer.fakeConnString)
-        )
-      );
-    }
+type AddSanitizer<T> = (
+  httpClient: HttpClient,
+  url: string,
+  recordingId: string,
+  sanitizer: T
+) => Promise<void>;
 
+/**
+ * Given an AddSanitizer<T> function, create an AddSanitizer function that operates on an array of T, adding
+ * each sanitizer in the array individually.
+ */
+const pluralize =
+  <T>(singular: AddSanitizer<T>): AddSanitizer<T[]> =>
+  async (httpClient: HttpClient, url: string, recordingId: string, sanitizers: T[]) => {
     await Promise.all(
-      (
-        [
-          // The following sanitizers have similar request bodies and this abstraction avoids duplication
-          "generalRegexSanitizers",
-          "bodyKeySanitizers",
-          "bodyRegexSanitizers",
-          "headerRegexSanitizers",
-          "uriRegexSanitizers",
-        ] as const
-      ).map((prop) => {
-        const replacers = options[prop];
-        if (replacers) {
-          return Promise.all(
-            replacers.map((replacer: RegexSanitizer) => {
-              if (
-                // sanitizers where the "regex" is a required attribute
-                [
-                  "bodyKeySanitizers",
-                  "bodyRegexSanitizers",
-                  "generalRegexSanitizers",
-                  "uriRegexSanitizers",
-                ].includes(prop) &&
-                !replacer.regex
-              ) {
-                if (!isRecordMode()) return;
-                throw new RecorderError(
-                  `Attempted to add an invalid sanitizer - ${JSON.stringify(replacer)}`
-                );
-              }
-              return this.addSanitizer({
-                sanitizer: sanitizerKeywordMapping[prop],
-                body: JSON.stringify(replacer),
-              });
-            })
-          );
-        } else return;
-      })
+      sanitizers.map((sanitizer) => singular(httpClient, url, recordingId, sanitizer))
     );
+  };
 
-    await Promise.all(
-      (
-        [
-          // The following sanitizers have similar request bodies and this abstraction avoids duplication
-          "resetSanitizer",
-          "oAuthResponseSanitizer",
-        ] as const
-      ).map((prop) => {
-        // TODO: Test
-        if (options[prop]) {
-          return this.addSanitizer({
-            sanitizer: sanitizerKeywordMapping[prop],
-            body: undefined,
-          });
-        } else return;
-      })
-    );
+/**
+ * Makes an AddSanitizer<unknown> function that passes the sanitizer content directly to the test proxy request body.
+ */
+const makeAddSanitizer =
+  (sanitizerName: ProxyToolSanitizers): AddSanitizer<Record<string, unknown>> =>
+  async (
+    httpClient: HttpClient,
+    url: string,
+    recordingId: string,
+    sanitizer: Record<string, unknown>
```

Comment:
Let's give this a better name?

`sanitizerBody`?

Reply:
I'd rather keep it consistent with the definition of `AddSanitizer` above. The name `sanitizerBody` doesn't make sense in the context of `AddSanitizer` since it suggests that it is going to be the body of the request, which is only the case with `makeAddSanitizer` here, and none of the other AddSanitizer functions.